config/default.go

```diff
@@ -4,49 +4,49 @@ import "reflect"
-		"username":                 &Entry{"", []any{}, reflect.String, false},
-		"password":                 &Entry{"", []any{}, reflect.String, false},
-		"options":                  &Entry{"", []any{}, reflect.String, false},
-		"maxOpenConnections":       &Entry{20, []any{}, reflect.Int, false},
-		"maxIdleConnections":       &Entry{20, []any{}, reflect.Int, false},
-		"maxLifetime":              &Entry{300, []any{}, reflect.Int, false},
-		"defaultReadQueryTimeout":  &Entry{20000, []any{}, reflect.Int, false},
-		"defaultWriteQueryTimeout": &Entry{40000, []any{}, reflect.Int, false},
+		"connection":               &Entry{"none", []any{}, reflect.String, false, false},
+		"host":                     &Entry{"127.0.0.1", []any{}, reflect.String, false, false},
+		"port":                     &Entry{0, []any{}, reflect.Int, false, false},
+		"name":                     &Entry{"", []any{}, reflect.String, false, false},
+		"username":                 &Entry{"", []any{}, reflect.String, false, false},
+		"password":                 &Entry{"", []any{}, reflect.String, false, false},
+		"options":                  &Entry{"", []any{}, reflect.String, false, false},
+		"maxOpenConnections":       &Entry{20, []any{}, reflect.Int, false, false},
+		"maxIdleConnections":       &Entry{20, []any{}, reflect.Int, false, false},
+		"maxLifetime":              &Entry{300, []any{}, reflect.Int, false, false},
+		"defaultReadQueryTimeout":  &Entry{20000, []any{}, reflect.Int, false, false},
+		"defaultWriteQueryTimeout": &Entry{40000, []any{}, reflect.Int, false, false},
 		"config": object{
-			"skipDefaultTransaction":                   &Entry{false, []any{}, reflect.Bool, false},
-			"dryRun":                                   &Entry{false, []any{}, reflect.Bool, false},
-			"prepareStmt":                              &Entry{true, []any{}, reflect.Bool, false},
-			"disableNestedTransaction":                 &Entry{false, []any{}, reflect.Bool, false},
-			"allowGlobalUpdate":                        &Entry{false, []any{}, reflect.Bool, false},
-			"disableAutomaticPing":                     &Entry{false, []any{}, reflect.Bool, false},
-			"disableForeignKeyConstraintWhenMigrating": &Entry{false, []any{}, reflect.Bool, false},
+			"skipDefaultTransaction":                   &Entry{false, []any{}, reflect.Bool, false, false},
+			"dryRun":                                   &Entry{false, []any{}, reflect.Bool, false, false},
+			"prepareStmt":                              &Entry{true, []any{}, reflect.Bool, false, false},
+			"disableNestedTransaction":                 &Entry{false, []any{}, reflect.Bool, false, false},
+			"allowGlobalUpdate":                        &Entry{false, []any{}, reflect.Bool, false, false},
+			"disableAutomaticPing":                     &Entry{false, []any{}, reflect.Bool, false, false},
+			"disableForeignKeyConstraintWhenMigrating": &Entry{false, []any{}, reflect.Bool, false, false},
```

Comment:
All of them except `server.proxy.host` should be required.

The validation is also performed when calling `config.Set()`. It is incorrect to set them to `nil` so it shouldn't be allowed.

Reply:
> This doesn't look like a breaking change to me if we drop the "empty" validation. It would only be breaking for codebases not always using keyed struct fields, which is very rare and usually not recommended anyway. 

Agreed. **But**, we can't dictate how people use the framework; There is risk if a new (minor or patch) release is done with a change to the contract for the `Entry`. It's also done without keyed fields in `config.default.go:73`. 

Your suggestion is to remove the 'empty', but what if in the future more keys need to be added? :)

Reply:
> All of them except `server.proxy.host` should be required.

Allright. But what if I don't want to set `"maxUploadSize": 10.0,` because I will never upload files and I would be fine with a default value of 10.0?

Reply:
> we can't dictate how people use the framework

Actually, we do since the framework is advertised as opinionated. Adding a field to a struct has never been considered a breaking change in any library I have contributed to.

> Your suggestion is to remove the 'empty', but what if in the future more keys need to be added? :)

I'm not sure I understand what you mean. Are you saying that we should think about potentially adding more struct fields to `Entry`?

> But what if I don't want to set "maxUploadSize": 10.0, because I will never upload files and I would be fine with a default value of 10.0?

The default value is 10 so it's not `nil`. Even if you don't provide it in your config file, the validation will not fail.

There are two scenarios:
- Default value is set to `nil` and entry is required: the developer **must** add it to their config file
- Default value is set to something and entry is required: the developer can, but is not forced to add it to their config file. However, they cannot set their value to `nil` (either from the config file or from `config.Set()`.

Reply:
> I'm not sure I understand what you mean. Are you saying that we should think about potentially adding more struct fields to `Entry`?

Not right now, but in general. One might add more fields to a struct. _Personally_ I think adding a constructor to a large struct is good in the long term. But... I'm not as experienced as you in Go so, if it's not considered a breaking change: good!

> The default value is 10 so it's not `nil`. Even if you don't provide it in your config file, the validation will not fail.

👍

Reply:
> All of them except `server.proxy.host` should be required.

Done ✅